packages/typespec-client-generator-core/lib/decorators.tsp

```diff
@@ -419,3 +419,19 @@ extern dec access(
  */
 #deprecated "@flattenProperty decorator is not recommended to use."
 extern dec flattenProperty(target: ModelProperty, scope?: valueof string);
+
+namespace Azure.ClientGenerator.Core.CSharp {
+  /**
+   * Whether a model needs the custom JSON converter, this is only used for backward compatibility.
+   * @param scope The language scope you want this decorator to apply to. If not specified, will only apply to csharp emitter.
+   *
+   * @example
+   * ```typespec
+   * @hasJSONConverter
+   * model MyModel {
+   *   prop: string;
+   * }
+   * ```
+   */
+  extern dec hasJSONConverter(target: Model, scope?: "csharp");
```

Comment:
I think `@hasJsonConverter` would be more correct with the naming standards we follow

Reply:
OK, renamed.